### ISSUE TITLE
Transposing Sparse Tensors in PyTorch

### ISSUE THREAD

**Issue by JunzheShen** [rllm-project/rllm#2]
![image](https://github.com/rllm-project/rllm/assets/104138466/6115cadb-024b-4bfb-a6ce-7da7f7a0c91c)
```
(Pdb) l
 69         adj_i, adj_v = adj.indices(), adj.values()
 70         hop = torch.zeros((dataset.v_num['movie'], dataset.v_num['movie'])).type(torch.LongTensor)
 71         for i in rating_range:
 72             idx = torch.where(adj_v == i, True, False)
 73             A = torch.sparse_coo_tensor(adj_i[:, idx], adj_v[idx], adj.shape)
 74 B->         A = torch.spmm(A.T, A).to_dense()
 75             hop |= torch.where(A > threshold, 1, 0)
 76         hop = hop.type(torch.FloatTensor)
 77  
 78         return dataset, \
 79                hop, \
(Pdb) p A.T
*** RuntimeError: Tensors of type SparseTensorImpl do not have strides
(Pdb) p A.transpose(0, 1)
tensor(indices=tensor([[3089, 3385, 2964,  ..., 3646, 3676, 3792],
                       [   1,    1,    2,  ..., 6039, 6039, 6039]]),
       values=tensor([1., 1., 1.,  ..., 1., 1., 1.]),
       size=(3883, 6040), nnz=56174, layout=torch.sparse_coo)
```
Hi!
When we are running the example code of GCN on movielens dataset, we encountered `RuntimeError: Tensors of type SparseTensorImpl do not have strides`. This is because the code here at `rllm/dataloader/movielens_classification.py:74` uses `A.T` to transpose a sparse matrix. Our PyTorch environment is 1.12.1+cu113, where `A.T` should be replaced by `A.transpose(0, 1)` (See figure above). Nevertheless, we can not rule out the possibility that this is due to some pytorch version issues. Thus, we recommend TAs provide some details about your environment, or modify the code.
Thank you UwU.

**Comment by Dlxxx** [comment]
! encountered the same problem

**Comment by rllm-project** [comment]
Fixed in latest push. This problem can also be fixed by upgrading torch to ver 2.2.0.
